packages/gatsby/src/commands/__tests__/built-utils.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  IGatsbyState,
+  IGatsbyPage,
+  IHtmlFileState,
+  IStaticQueryResultState,
+} from "../../redux/types"
+import { calcDirtyHtmlFiles } from "../build-utils"
+
+interface IMinimalStateSliceForTest {
+  html: IGatsbyState["html"]
+  pages: IGatsbyState["pages"]
+}
+
+describe(`calcDirtyHtmlFiles`, () => {
+  function generateStateToTestHelper(
+    pages: Record<
+      string,
+      {
+        dirty: number
+        removedOrDeleted?: "deleted" | "not-recreated"
+      }
+    >
+  ): IGatsbyState {
+    const state: IMinimalStateSliceForTest = {
+      pages: new Map<string, IGatsbyPage>(),
+      html: {
+        browserCompilationHash: `a-hash`,
+        ssrCompilationHash: `a-hash`,
+        trackedHtmlFiles: new Map<string, IHtmlFileState>(),
+        unsafeBuiltinWasUsedInSSR: false,
+        trackedStaticQueryResults: new Map<string, IStaticQueryResultState>(),
+      },
+    }
+
+    for (const pagePath in pages) {
+      const page = pages[pagePath]
+
+      if (page.removedOrDeleted !== `not-recreated`) {
+        state.pages.set(pagePath, {
+          component: `/foo`,
+          componentPath: `/foo`,
+          componentChunkName: `foo`,
+          context: {},
+          internalComponentName: `foo`,
+          isCreatedByStatefulCreatePages: false,
+          path: pagePath,
+          matchPath: undefined,
+          pluginCreatorId: `foo`,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          pluginCreator___NODE: `foo`,
+          updatedAt: 1,
+        })
+      }
+
+      state.html.trackedHtmlFiles.set(pagePath, {
+        dirty: page.dirty,
+        pageDataHash: `a-hash`,
+        isDeleted: page.removedOrDeleted === `deleted`,
+      })
+    }
+
+    return state as IGatsbyState
+  }
+
+  it(`nothing changed`, () => {
+    const state = generateStateToTestHelper({
+      "/": {
+        dirty: 0,
+      },
+    })
+
+    const results = calcDirtyHtmlFiles(state)
+    expect(results).toMatchInlineSnapshot(`
+      Object {
+        "toCleanupFromTrackedState": Set {},
+        "toDelete": Array [],
+        "toRegenerate": Array [],
+      }
+    `)
```

Comment:
I will adjust tests to not use snapshots as that's often foot gun - I just pushed current state of those